pkg/registry/core/rest/storage_core.go

```diff
@@ -92,7 +92,7 @@ type rangeRegistries struct {
 	nodePort           rangeallocation.RangeRegistry
 }
 
-type legacyProvider struct {
+type StorageProvider struct {
```

Comment:
do you need this in another split out-PR?

Reply:
Oh, I think I can revert it now. Was need before for some test.